src/Parsing/Parser.hs

```diff
@@ -189,3 +202,139 @@ someASTParser PHP        = Just (SomeASTParser (ASTParser tree_sitter_php :: Par
+--
+-- In practice, this means using 'Control.Effect.Parse.parseWith', and passing in a map of parsers to select from for your feature. It is recommended to define the map as a concrete top-level binding using the abstract parsers or ideally the canonical maps of parsers, below; using the abstracted parsers or canonical maps directly with 'Control.Effect.Parse.parseWith' will lead to significantly slower compiles.
+--
+-- Bad:
+--
+-- @
+-- isFancy :: (Carrier sig m, Member Parse sig) => Blob -> m Bool
+-- isFancy = parseWith (preciseParsers @Fancy) (pure . isTermFancy) -- slow compiles!
+-- @
+--
+-- Good:
+--
+-- @
+-- fancyParsers :: 'Map' 'Language' ('SomeParser' Fancy 'Loc')
+-- fancyParsers = preciseParsers
+--
+-- isFancy :: (Carrier sig m, Member Parse sig) => Blob -> m Bool
+-- isFancy = parseWith fancyParsers (pure . isTermFancy) -- much faster compiles
+-- @
+
+
+-- | A parser producing terms of existentially-quantified type under some constraint @c@.
+--
+--   This can be used to perform actions on terms supporting some feature abstracted using a typeclass, without knowing (or caring) what the specific term types are.
+data SomeParser c a where
+  SomeParser :: c t => Parser (t a) -> SomeParser c a
+
+goParser' :: c (Term (Sum Go.Syntax)) => (Language, SomeParser c Loc)
+goParser' = (Go, SomeParser goParser)
+
+haskellParser' :: c (Term (Sum Haskell.Syntax)) => (Language, SomeParser c Loc)
+haskellParser' = (Haskell, SomeParser haskellParser)
+
+javascriptParser' :: c (Term (Sum TSX.Syntax)) => (Language, SomeParser c Loc)
+javascriptParser' = (JavaScript, SomeParser tsxParser)
+
+jsonParser' :: c (Term (Sum JSON.Syntax)) => (Language, SomeParser c Loc)
+jsonParser' = (JSON, SomeParser jsonParser)
+
+jsxParser' :: c (Term (Sum TSX.Syntax)) => (Language, SomeParser c Loc)
+jsxParser' = (JSX, SomeParser tsxParser)
+
+markdownParser' :: c (Term (Sum Markdown.Syntax)) => (Language, SomeParser c Loc)
+markdownParser' = (Markdown, SomeParser markdownParser)
+
+phpParser' :: c (Term (Sum PHP.Syntax)) => (Language, SomeParser c Loc)
+phpParser' = (PHP, SomeParser phpParser)
+
+pythonParserALaCarte' :: c (Term (Sum Python.Syntax)) => (Language, SomeParser c Loc)
+pythonParserALaCarte' = (Python, SomeParser pythonParser)
+
+pythonParserPrecise' :: c Py.Term => (Language, SomeParser c Loc)
+pythonParserPrecise' = (Python, SomeParser pythonParserPrecise)
+
+pythonParser' :: (c (Term (Sum Python.Syntax)), c Py.Term) => PerLanguageModes -> (Language, SomeParser c Loc)
+pythonParser' modes = case pythonMode modes of
+  ALaCarte -> (Python, SomeParser pythonParser)
+  Precise  -> (Python, SomeParser pythonParserPrecise)
```

Comment:
When we’re finished migrating support for Python over to the precise stuff, then this will go away and we’ll only have the precise version ✨